QuestionAnswering/QA_ICL/backend/fastapi_app/README.md

```diff
@@ -54,42 +87,47 @@ ontomops_fileserver:
 
 Precedence: `app.local.yaml` > `app.{APP_ENV}.yaml` > `app.yaml`.
 
+
 ## Native installation
 
 ### Prerequisites
 
-- (recommended) Linux enviroment, including WSL
+- (recommended) Linux enviroment
 - python>=3.8
 - [conda](https://conda.io/projects/conda/en/latest/user-guide/install/index.html)
-
   
 ### Steps
 
-1. Create conda environment and activate it.
+1. Deploy the required services, add the resources, and configure the parameters as defined in the [Initial setup](#initial-setup) section.
+
+2. Create conda environment and activate it.
    ```{bash}
    conda create --name qa_backend python==3.10
    conda activate qa_backend
    ```
 
-2. Install dependencies.
+3. Install dependencies.
    ```{bash}
    pip install -r requirements.txt
    ```
 
-3. Ingest required resources into Redis server, as per the section ['Required resources'](#required-resources).
+4. Ingest required datasets into Redis server, as per the section [Required datasets](#required-datasets).
 
-4. Start the server:
+5. Start the server:
    - In debug mode (app is automatically reloaded upon code changes), `uvicorn main:app --reload --log-config=log_conf.yaml`.
    - In production mode, `uvicorn main:app --host=0.0.0.0 --log-config log_conf.yaml --workers 4`.
-   - The app will be available at `localhost:8000`. 
+   - The app will be available at `localhost:8000`.  
+   <!-- TODO: add a health check endpoint -->
    - To expose the app at a different port, use the command line argument `--port {port}` e.g. `uvicorn main:app --reload --log-config=log_conf.yaml --port 5000`.
 
-5. Whenever any resource needs to be updated, re-run the ingestion script for that specific resource with the argument `--drop_index --invalidate_cache` to (1) trigger Redis to flush the old index and create a new one, and (2) re-create on-disk cache for the processed resource. KIV: allow user to add new resource entries without processing everything again and recreatng the index.
+6. Whenever any dataset needs to be updated, re-run the ingestion script for that specific dataset with the argument `--drop_index --invalidate_cache` to (1) trigger Redis to flush the old index and create a new one, and (2) re-create on-disc cache for the processed datasets. KIV: allow user to add new resource entries without processing everything again and recreatng the index.
+
+## Docker installation
 
-## Dockerised installation
+1. Ensure that the required services are deployed, resources are added, and parameters are configured as specified in the [Initial setup](#initial-setup) section.
 
 1. Build the image, `docker build -t fastapi_app .`.
-1. Spin up the container as follows. The app will be available at `localhost:5000` on Docker host.
+2. Spin up the container as follows. The app will be available at `localhost:5000` on Docker host.
    ```{bash}
    docker run --name fastapi_app \
```

Comment:
Again, I believe we're almost certainly missing a `-d`, otherwise the console will lock.

Reply:
The `-d` flag has been added to the `docker run` command.